draft-packages/likert-scale-legacy/KaizenDraft/LikertScaleLegacy/LikertScaleLegacy.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useState, createRef } from "react"
+
+import classnames from "classnames"
+import { Paragraph } from "@kaizen/component-library"
+import determineSelectionFromKeyPress from "./helpers/determineSelectionFromKeyPress"
+import { Scale, ScaleItem, ScaleValue } from "./types"
+
+const styles = require("./styles.module.scss")
+
+type ItemRefs = Array<{
+  value: ScaleValue
+  ref: { current: null | HTMLDivElement }
+}>
+
+export interface LikertScaleProps {
+  labelId: string
+  scale: Scale
+  selectedItem: ScaleItem | null
+  automationId: string
+  reversed?: boolean
+  validationError?: boolean
+  validationErrorId?: string
+  onSelect: (value: ScaleItem | null) => void
+}
+
+export const LikertScaleLegacy = ({
+  scale,
+  selectedItem,
+  reversed,
+  automationId,
+  onSelect,
+  validationError,
+  validationErrorId,
+  labelId,
+}: LikertScaleProps) => {
+  const [hoveredItem, setHoveredItem] = useState<ScaleItem | null>(null)
+  const itemRefs: ItemRefs = scale.map(s => ({
+    value: s.value,
+    ref: createRef<HTMLDivElement>(),
+  }))
+
+  const handleRadioClick = (item: ScaleItem): void => {
+    const newValue =
+      selectedItem && item.value === selectedItem.value
+        ? scale.find(s => s.value === -1) || null
+        : item
+    onSelect(newValue)
+
+    // Remove hover state
+    setHoveredItem(null)
+  }
+
+  const handleRadioMouseEnter = (item: ScaleItem): void => {
+    if (selectedItem && selectedItem.value > 0) {
+      return // Disable hover when there is a selection
+    }
+    setHoveredItem(item)
+  }
+
+  const handleRadioMouseLeave = (): void => {
+    if (selectedItem && selectedItem.value > 0) {
+      return // Disable hover when there is a selection
+    }
+    setHoveredItem(null)
+  }
+
+  /**
+   * Because the radios have been built with divs, we need to add the keyboard functionality manually
+   */
+  const handleKeyDown = (
+    event: React.KeyboardEvent<HTMLDivElement>,
+    focusedItem: ScaleItem
+  ): void => {
+    const newPosition = determineSelectionFromKeyPress(
+      event.keyCode,
+      selectedItem,
+      focusedItem
+    )
+    if (newPosition) {
+      event.preventDefault()
+
+      onSelect(scale.find(s => s.value === newPosition) || null)
+
+      // Update focus
+      const itemRef = itemRefs.find(item => item.value === newPosition)
+      itemRef && itemRef.ref.current && itemRef.ref.current.focus()
+    }
+  }
+
+  let legend = "Not rated"
+  if (hoveredItem) {
+    legend = hoveredItem.label
+  } else if (selectedItem) {
+    legend = selectedItem.label
+  }
+
+  return (
+    <div
+      className={classnames(styles.container, {
+        [styles.rated]: selectedItem && selectedItem.value > 0,
+        [styles.reversed]: reversed,
+      })}
+      aria-labelledby={labelId}
+      role="radiogroup"
+      tabIndex={-1}
+      aria-describedby={(validationError && validationErrorId) || undefined}
```

Comment:
Can we please add `data-automation-id={automationId}` here, in case people want to select on the container.